### PR TITLE
Support identity map in setConsent

### PR DESCRIPTION
**Pull request by jonsnyder**
## Description

Add an identityMap option in setConsent, and pass to Konductor.  Consolidate the identityMap validation between sendEvent and setConsent.  Add functional test.

## Related Issue

https://jira.corp.adobe.com/browse/CORE-47162

## Motivation and Context

Many times setConsent is the call the establishes identity.  If customers are providing their own id, that should be used to create the ecid.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

### CODE REVIEW

File: src/components/Privacy/validateSetConsentOptions.js [hunk 1]
```diff
@@ -38,7 +39,20 @@ export default objectOf({
...
+  identityMap: mapOfValues(
+    arrayOf(
+      objectOf({
+        authenticatedState: string(),
+        id: string(),
+        namespace: objectOf({
+          code: string()
+        }).noUnknownFields(),
+        primary: boolean(),
+        xid: string()
```
Comment (jfkhoury) [inline src/components/Privacy/validateSetConsentOptions.js]: What is `xid`?

File: test/functional/specs/Privacy/C225953.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import createFixture from "../../helpers/createFixture";
```
Comment (jfkhoury) [inline test/functional/specs/Privacy/C225953.js:1]: Nice!

File: src/components/Privacy/validateSetConsentOptions.js (hunk 1 above)
Comment (jonsnyder) [reply src/components/Privacy/validateSetConsentOptions.js]: https://github.com/adobe/xdm/blob/master/schemas/context/identity.schema.json

File: src/components/DataCollector/validateUserEventOptions.js [hunk 3]
```diff
@@ -10,19 +10,8 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
-  enumOf,
-  mapOfValues
-} from "../../utils/validation";
-import {
-  AMBIGUOUS,
-  AUTHENTICATED,
-  LOGGED_OUT
-} from "../../constants/identityMapAuthenticatedState";
+import { string, objectOf, boolean, arrayOf } from "../../utils/validation";
+import { validateIdentityMap } from "../../utils";
```
Comment (liljenback) [inline src/components/DataCollector/validateUserEventOptions.js:14]: 👍

File: src/utils/validateIdentityMap.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      authenticatedState: enumOf(AMBIGUOUS, AUTHENTICATED, LOGGED_OUT),
+      id: string(),
+      namespace: objectOf({
+        code: string()
+      }).noUnknownFields(),
+      primary: boolean(),
+      xid: string()
+    })
+  )
+    .nonEmpty()
```
Comment (liljenback) [inline src/utils/validateIdentityMap.js:39]: Do we follow these restrictions on other API calls? Is it too restrictive?

Comment (jonsnyder) [reply src/utils/validateIdentityMap.js:39]: That's a good point. I think we should allow empty arrays

**Pull request merged**